### PR TITLE
Fix memory leaks

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintTrafficLightAction.kt [hunk 1]
```diff
@@ -51,13 +51,16 @@ class SonarLintTrafficLightAction(private val editor: Editor) : AbstractSonarAct
...
+            return
+        }
         e.getData(CommonDataKeys.VIRTUAL_FILE)?.let { file ->
             val presentation = e.presentation
             val isFocusOnNewCode = getService(project, CleanAsYouCodeService::class.java).shouldFocusOnNewCode(project)
             val relevantFindings = getService(project, FindingsCache::class.java).getFindingsForFile(file).filter { !it.isResolved && (!isFocusOnNewCode || it.isOnNewCode()) }
             val issues = relevantFindings.filterIsInstance<LiveIssue>()
             val hotspots = relevantFindings.filterIsInstance<LiveSecurityHotspot>()
-            val taintVulnerabilitiesCount = TaintVulnerabilitiesCache.getStatus(project)?.count(isFocusOnNewCode) ?: 0
+            val taintVulnerabilitiesCount = getService(project, TaintVulnerabilitiesCache::class.java).status?.count(isFocusOnNewCode) ?: 0
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintTrafficLightAction.kt:63]: Unrelated to this PR, but it made me think about whether we should check if the project is disposed inside the `getService(project, class)` method?
There can be errors like 
```
	at com.intellij.serviceContainer.ContainerUtilKt.throwAlreadyDisposedError(containerUtil.kt:45)
	at com.intellij.serviceContainer.ComponentManagerImpl.checkThatCreatingOfLightServiceIsAllowed(ComponentManagerImpl.kt:790)
	at com.intellij.serviceContainer.ComponentManagerImpl.getOrCreateLightService(ComponentManagerImpl.kt:770)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:726)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:670)
	at org.sonarlint.intellij.common.util.SonarLintUtils.getService(SonarLintUtils.java:62)
```

File: src/main/java/org/sonarlint/intellij/trigger/BuildFinishedAnalysisTrigger.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+public class BuildFinishedAnalysisTrigger implements BuildManagerListener {
+
+  @Override public void buildFinished(Project project, UUID sessionId, boolean isAutomake) {
+    if (!isAutomake) {
+      // covered by CompilationFinishedAnalysisTrigger.compilationFinished
+      return;
+    }
+
+    SonarLintUtils.getService(project, SonarLintConsole.class).debug("build finished");
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/trigger/BuildFinishedAnalysisTrigger.java:37]: is it worth checking for project disposed?

**Review by nquinquenel — APPROVED** [review]
The changes look good 🤞🏻

File: src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintTrafficLightAction.kt (hunk 1 above)
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/ui/traffic/light/SonarLintTrafficLightAction.kt:63]: This is one thing I wanted to explore, I just created [SLI-1236](https://sonarsource.atlassian.net/browse/SLI-1236) after a community report (and I also experienced it myself). Should we have a generic way to deal with disposed services? I will think about it when working on this ticket

File: src/main/java/org/sonarlint/intellij/trigger/BuildFinishedAnalysisTrigger.java (hunk 2 above)
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/trigger/BuildFinishedAnalysisTrigger.java:37]: I don't know. We were not checking before. Same as my previous comment, let me check first what should be the approach for checking disposed projects

**Pull request merged**